add: sanitise query params in remote logging (PHP)

rjchow opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Related to https://github.com/woocommerce/woocommerce/issues/50984.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch
3. Go to `Tools -> WCA Test Helper -> Remote Logging`
4. Enable remote logging
5. Click on PHP `Simulate Core Exception`, navigate to a wc-admin url with some arbitrary query params such as: `wp-admin/admin.php?page=wc-settings&tab=general&key=very_secret_key`
7. Check the log server page for the log entry (See test plan in D155228-code to see how to check the log entry)
8. Confirm the log properties include `request_uri` and the non-whitelisted parameters have been masked


<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

chihsuan reviewed (approved):
Thanks for working on this! @rjchow LGTM and tested well. 💯 

Should we also make similar changes in the client side? 🤔

moon0326 commented on plugins/woocommerce/src/Internal/Logging/RemoteLogger.php:
```diff
@@ -435,4 +435,52 @@ function ( $trace_item ) {
+
+		foreach ( $query_params as $key => &$value ) {
+			if ( ! in_array( $key, $whitelist, true ) ) {
+				$value = 'xxxxxx';
```
nitpic: May be we can just set it empty?

rjchow replied:
I think it's valuable to be able to distinguish between a masked value and an actually empty value, since empty values can be the source of bugs. I don't really like 'xxxxxx' but I don't have a better idea for a special, distinguished value 🤣

moon0326 reviewed (approved):
Left a minor comment, but LGTM 👍 🚀

rjchow committed:
feat: add request URI sanitization to log only whitelisted parameters

test: add unit tests for sanitize_request_uri method in RemoteLogger class

feat: add WordPress filter to modify request URI whitelist and implement corresponding tests

refactor: extract whitelist array into a separate variable for improved readability

refactor: rename variable for clarity in sanitize_request_uri method

refactor: rename request URI whitelist filter to match existing naming conventions

test: Remove unnecessary filter removal in RemoteLoggerTest class

test: add test for request URI sanitization in get_formatted_log method

test: add null check for request_uri in RemoteLoggerTest

fixed tests

changelog

php lint fix:

added tab to query param whitelist

updated doc
